kernel/src/gdt.c

```diff
@@ -97,6 +97,9 @@ void setup_gdt_and_tss() {
     // Allocate privilege level 0 stack
     g_tss.rsp[0] = (void*)alloc_pages(TSS_STACK_PAGES, PAGING_WRITABLE);
 
+    // Move stack to end of allocation
+    g_tss.rsp[0] += TSS_STACK_PAGES * PAGE_SIZE;
```

Comment:
You should probably just do this on the line where the stack is being allocated instead of splitting it in two. 

The IST pointer should also start at the bottom as well.